aws/connection_config.go

```diff
@@ -5,59 +5,28 @@ import (
 	"strings"
 
 	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
-	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/schema"
 )
 
 type awsConfig struct {
-	Regions               []string `cty:"regions"`
-	DefaultRegion         *string  `cty:"default_region"`
-	Profile               *string  `cty:"profile"`
-	AccessKey             *string  `cty:"access_key"`
-	SecretKey             *string  `cty:"secret_key"`
-	SessionToken          *string  `cty:"session_token"`
-	MaxErrorRetryAttempts *int     `cty:"max_error_retry_attempts"`
-	MinErrorRetryDelay    *int     `cty:"min_error_retry_delay"`
-	IgnoreErrorCodes      []string `cty:"ignore_error_codes"`
-	EndpointUrl           *string  `cty:"endpoint_url"`
-	S3ForcePathStyle      *bool    `cty:"s3_force_path_style"`
+	Regions               []string          `hcl:"regions,optional"`
```

Comment:
@brittandeyoung Referencing your most recent issue comment https://github.com/turbot/steampipe-plugin-aws/issues/1844#issuecomment-1722562384, you mention using `cty`, but here in the code I see all fields using `hcl` instead.

How does the code currently behave? Does it not work in some scenarios?

Reply:
@cbruno10 In the current state, the `role_arn` loads in properly and assuming the role works as expected as it has the `cty` tag. 

Currently the remainder of the fields to not have the `cty` tag. Without this tag, these items DO NOT load in from the config. This means these values are `nil`.

If I add the `cty` tag, the values do load in, but then are no longer optional and must be provided. If you do not provide them all, steampipe crashes and fails to load. If i attempt to add the `optional` tag, steampipe crashes when attempting to load the config. 

Any tips on direction from here is appreciated

Reply:
We were able to reproduce the above issue - 
```
Error: failed to start plugin 'hub.steampipe.io/plugins/turbot/aws@latest': failed to start 'hub.steampipe.io/plugins/turbot/aws@latest': failed to parse connection config for connection 'aws':
Unsuitable value type: Unsuitable value: attribute "external_id" is required
```

This could be an HCL parsing error - https://github.com/turbot/steampipe-plugin-sdk/blob/main/plugin/connection_config_schema.go#L88

I have raised a Steampipe SDK [issue](https://github.com/turbot/steampipe-plugin-sdk/issues/692) for the same with the details.

Reply:
@brittandeyoung, please let us know if you are getting any other issues than the above.

Reply:
@bigdatasourav Great thanks! That was the only issue i was running into. I will keep an eye on this issue.